cli/src/pcluster/templates/cluster_stack.py

```diff
@@ -302,35 +304,64 @@ def _add_resources(self):
     def _add_alarms(self):
         self.alarms = []
 
-        metrics_for_alarms = {
-            "Mem": cloudwatch.Metric(
-                namespace="CWAgent",
-                metric_name="mem_used_percent",
-                dimensions_map={"InstanceId": self.head_node_instance.ref},
-                statistic="Maximum",
-                period=Duration.seconds(CW_ALARM_PERIOD_DEFAULT),
-            ),
-            "Disk": cloudwatch.Metric(
-                namespace="CWAgent",
-                metric_name="disk_used_percent",
-                dimensions_map={"InstanceId": self.head_node_instance.ref, "path": "/"},
-                statistic="Maximum",
-                period=Duration.seconds(CW_ALARM_PERIOD_DEFAULT),
-            ),
+        alarm_settings = {
+            "Mem": {
+                "namespace": "CWAgent",
+                "metric_name": "mem_used_percent",
+                "dimensions_map": {"InstanceId": self.head_node_instance.ref},
+                "statistic": "Maximum",
+                "threshold": CW_ALARM_PERCENT_THRESHOLD_DEFAULT,
+                "comparison_operator": cloudwatch.ComparisonOperator.GREATER_THAN_THRESHOLD,
+            },
+            "Disk": {
+                "namespace": "CWAgent",
+                "metric_name": "disk_used_percent",
+                "dimensions_map": {"InstanceId": self.head_node_instance.ref, "path": "/"},
+                "statistic": "Maximum",
+                "threshold": CW_ALARM_PERCENT_THRESHOLD_DEFAULT,
+                "comparison_operator": cloudwatch.ComparisonOperator.GREATER_THAN_THRESHOLD,
+            },
         }
 
-        for metric_key, metric in metrics_for_alarms.items():
-            alarm_id = f"HeadNode{metric_key}Alarm"
-            alarm_name = f"{self.stack.stack_name}_{metric_key}Alarm_HeadNode"
+        if self._condition_is_slurm() and is_feature_supported(Feature.CLUSTER_HEALTH_METRICS):
+            alarm_settings["ProtectedMode"] = {
+                "namespace": "ParallelCluster",
+                "metric_name": "ClusterInProtectedMode",
+                "dimensions_map": {"ClusterName": self.stack.stack_name},
+                "period": Duration.seconds(CW_ALARM_PERIOD_DEFAULT),
+                "threshold": PROTECTED_MODE_THRESHOLD,
+                "comparison_operator": cloudwatch.ComparisonOperator.GREATER_THAN_OR_EQUAL_TO_THRESHOLD,
+            }
+
+        for alarm_key, alarm_values in alarm_settings.items():
+            alarm_id = f"HeadNode{alarm_key}Alarm"
+            alarm_name = f"{self.stack.stack_name}_{alarm_key}Alarm_HeadNode"
+
+            if alarm_key == "ProtectedMode":
```

Comment:
Instead of adding if else here, what about having `statistic=alarm_values.get("statistic")`?

Reply:
Thanks for the advice! Revised!